lightningd/signmessage.c

```diff
@@ -137,6 +138,12 @@ static void listnodes_done(const char *buffer,
 	if (t)
 		t = json_get_member(buffer, t, "nodes");
 
+	if (!deprecated_apis && (!t || t->size == 0)) {
+		was_pending(command_fail(can->cmd, SIGNMESSAGE_PUBKEY_NOT_FOUND,
+					 "pub key not found in the graph, expected pubkey is %s",
```

Comment:
We can actually include this as structured data, rather than a string (and "pub key" is one word!):

```
	struct json_stream *response;
	response = json_stream_fail(cmd, SIGNMESSAGE_PUBKEY_NOT_FOUND,
					    "pubkey not found in the graph");
	json_add_node_id(response, "claimed_key", &can->id);
	json_object_end(response);
	return command_failed(cmd, response);
```

Reply:
Sound good! the only thing that I did not catch is the way we need to change the function type and return an error?

I assumed that this was only an example and I make a partial application of your suggestion.

Let me know if I'm missing somethings